### ISSUE TITLE
Model Degradation During Fine-Tuning on Mesh Representation with Depth and Normal Maps

### ISSUE THREAD

**Issue by SBlumenstock1** [TencentARC/InstantMesh#149]
I'm experiencing a degradation in model performance when fine-tuning with mesh representation (depth and normal maps). The model initially works well when fine-tuned on the NeRF representation but begins to degrade after a few steps when fine-tuning with mesh-based training. This issue affects depth, normal maps, and the overall 3D reconstruction quality, as seen in the attached training logs.

<p align="center">
  <img src="https://github.com/user-attachments/assets/faf5196a-1944-4623-a62d-fde602446450" width="200"/>
  <img src="https://github.com/user-attachments/assets/7e224aa0-1f93-44a9-8b8b-9d3a8c5aa2db" width="200"/>
  <img src="https://github.com/user-attachments/assets/d49b8509-fddc-4e97-8aaf-7214e3c2b25c" width="200"/>
  <img src="https://github.com/user-attachments/assets/687a1895-7ecc-4573-8d4a-4f46428ae0be" width="200"/>
</p>

It seems like the loss for geometry-related aspects (depth, normals) might not be contributing adequately, or the SDF regularization is too aggressive, leading to collapsing geometry.
Any insights or suggestions would be greatly appreciated!

**Comment by feiqiu-cyber** [comment]
> I'm experiencing a degradation in model performance when fine-tuning with mesh representation (depth and normal maps). The model initially works well when fine-tuned on the NeRF representation but begins to degrade after a few steps when fine-tuning with mesh-based training. This issue affects depth, normal maps, and the overall 3D reconstruction quality, as seen in the attached training logs.
> 
> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/166011324/365084441-faf5196a-1944-4623-a62d-fde602446450.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMDEwNjAsIm5iZiI6MTczMjAwMDc2MCwicGF0aCI6Ii8xNjYwMTEzMjQvMzY1MDg0NDQxLWZhZjUxOTZhLTE5NDQtNDYyMy1hNjJkLWZkZTYwMjQ0NjQ1MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOVQwNzE5MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YmNmNzYyMjA2ZTg3Yjk0NzM3ODJmMTM4YzU4MTlkMzEzODZlNjQyYjJmNDU5OTUwODExOGZmN2Q4ZjIxNjcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.uDt_oBCkloRVTzbGWSRn4afIJPgtF4_62nB3copjy70"> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/166011324/365084458-7e224aa0-1f93-44a9-8b8b-9d3a8c5aa2db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMDEwNjAsIm5iZiI6MTczMjAwMDc2MCwicGF0aCI6Ii8xNjYwMTEzMjQvMzY1MDg0NDU4LTdlMjI0YWEwLTFmOTMtNDRhOS04YjhiLTlkM2E4YzVhYTJkYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOVQwNzE5MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZTg2YjQ0NjcwZGI0OWIxZTIyZWJmNTA5MDQwZTBjMDc0YWQyYzU3YTI4NjMwNTc1OTM1YWY4ZjA0NjkwMmQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.huK96PT1ngtq_Jr058A6runU8si1vx5pDf6ax3NyrcI"> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/166011324/365084466-d49b8509-fddc-4e97-8aaf-7214e3c2b25c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMDEwNjAsIm5iZiI6MTczMjAwMDc2MCwicGF0aCI6Ii8xNjYwMTEzMjQvMzY1MDg0NDY2LWQ0OWI4NTA5LWZkZGMtNGU5Ny04YWFmLTcyMTRlM2MyYjI1Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOVQwNzE5MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDJlYzBlNDlmYWRmMzU2YzIyNjQ2NGU2NThiZDljYjc2NzJmODA2Y2RmOWM1YzY4ODZmMzA2MzVjOTM4MDBmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.hUeDT2ZLZGsO5tfaG-goV7IRibt5yKp0pVvX95jl5vs"> <img alt="" width="200" src="https://private-user-images.githubusercontent.com/166011324/365084469-687a1895-7ecc-4573-8d4a-4f46428ae0be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMDEwNjAsIm5iZiI6MTczMjAwMDc2MCwicGF0aCI6Ii8xNjYwMTEzMjQvMzY1MDg0NDY5LTY4N2ExODk1LTdlY2MtNDU3My04ZDRhLTRmNDY0MjhhZTBiZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExOVQwNzE5MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzI0OGU0OWFlZTU3MTc4ODI5N2YzNDhmN2E4YTdhOTk5MTE2YjBiYzE5NjE4MTRlOGM4YjA2MTNkYjE1ODAxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._rfYyJ125iY5IMswblYsD5yIRmnma0puA2vM2uwgun0">
> 
> It seems like the loss for geometry-related aspects (depth, normals) might not be contributing adequately, or the SDF regularization is too aggressive, leading to collapsing geometry. I have tested to remove the loss_reg and the error has disappeared, so the problem is within the calculation of the flexicubes regularization loss.
> 
> Any insights or suggestions would be greatly appreciated!

Hello, I met the same problem as you, have you solved it?

**Comment by ipekoztas** [comment]
Hi, how did you generate the normal maps?

**Comment by grrzx** [comment]
Hello, I also encountered the same issue when fine-tuning InstantMesh - the results at step=0 are decent, but the quality degrades as training progresses, even leading to reconstruction collapse. I tried freezing partial parameters and adjusting learning rates, but without success. I think your suggestion about dropping the reg loss is a promising direction.

Could you share more details about your approach? For instance:

Did you perform full fine-tuning or freeze specific components (e.g., encoder, decoder, latent space)?
How did you modify the regularization loss - by completely removing certain terms or adjusting weight coefficients?
Were there any other key modifications to the training pipeline (optimizer settings, gradient clipping, etc.)?
Any implementation details or empirical observations would be greatly appreciated!
